extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.oidc.runtime;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.quarkus.oidc.OidcConfig;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public final class OidcUtils {
+    private OidcUtils() {
+
+    }
+
+    public static List<String> findRoles(OidcConfig.Roles rolesConfig, JsonObject json) throws Exception {
+        // If the user configured a specific path - check and enforce a claim at this path exists
+        if (rolesConfig.getRoleClaimPath().isPresent()) {
+            return findClaimWithRoles(rolesConfig, rolesConfig.getRoleClaimPath().get(), json, true);
+        }
+
+        // Check 'groups' next
+        List<String> groups = findClaimWithRoles(rolesConfig, "groups", json, false);
+        if (!groups.isEmpty()) {
+            return groups.stream().map(v -> v.toString()).collect(Collectors.toList());
+        } else {
+            // Finally, check if this token has been issued by Keycloak
+            return findClaimWithRoles(rolesConfig, "realm_access/roles", json, false);
```

Comment:
Could you also include client roles in addition to realm ones?

Client roles follow a format like this:

```json
{
  "resource_access": {
    "frontend": {
      "roles": [
        "r1"
      ]
    },
}
```

Reply:
@pedroigor sure, I'll give it a try

Reply:
It is the accumulation, both realm and resource access roles are supposed to be checked ?

Reply:
@pedroigor Have a look please, I've updated the code to have both `realm_access` and `resource_access` collected